tests/Integration/Behaviour/Features/Context/Domain/ContactFeatureContext.php

```diff
@@ -0,0 +1,145 @@
+
+class ContactFeatureContext extends AbstractDomainFeatureContext
+{
+    private const DEFAULT_LOCALE_ID = 1; // EN locale
+    private const DUMMY_CONTACT_ID = 1;
+
+    /**
+     * @var int
+     */
+    private $defaultLangId;
+
+    public function __construct()
+    {
+        $configuration = CommonFeatureContext::getContainer()->get('prestashop.adapter.legacy.configuration');
+        $this->defaultLangId = $configuration->get('PS_LANG_DEFAULT');
+    }
+
+    /**
+     * @Given there is no contact with id :contactId
+     *
+     * @param int $contactId
+     */
+    public function thereIsNoContactWithId(int $contactId)
+    {
+        try {
+            $this->getQueryBus()->handle(new GetContactForEditing($contactId));
+        } catch (ContactNotFoundException $exception) {
+        }
+    }
+
+    /**
+     * @Given there is contact with id :contactId
+     *
+     * @param int $contactId
+     */
+    public function thereIsContactIsWithId(int $contactId)
+    {
+        $this->getQueryBus()->handle(new GetContactForEditing($contactId));
+    }
+
+    /**
+     * @Then I should be able to get contact with id :contactId for editing
+     *
+     * @param int $contactId
+     */
+    public function iShouldBeAbleToGetContactWithIdForEditing(int $contactId)
+    {
+        $this->getQueryBus()->handle(new GetContactForEditing($contactId));
+    }
+
+    /**
+     * @When I add new contact with the following properties:
+     *
+     * @param TableNode $table
+     */
+    public function iAddNewContactWithTheFollowingProperties(TableNode $table)
+    {
+        $data = $this->extractFirstHorizontalRowFromProperties($table);
+        /** @var EditableContact $editablContact */
+        $editableContact = $this->mapToEditableContact(self::DUMMY_CONTACT_ID, $data);
+
+        $addContactCommand = new AddContactCommand(
+            $editableContact->getLocalisedTitles(), $editableContact->isMessagesSavingEnabled()
+        );
+        $addContactCommand->setEmail($editableContact->getEmail()->getValue());
+        $addContactCommand->setLocalisedDescription($editableContact->getLocalisedDescription());
+        $addContactCommand->setShopAssociation($editableContact->getShopAssociation());
+
+        $this->getCommandBus()->handle($addContactCommand);
+    }
+
+    /**
+     * @When contact with id :contactId should have the following properties:
+     *
+     * @param int $contactId
+     * @param TableNode $table
+     */
+    public function contactWithIdShouldHaveTheFollowingProperties(int $contactId, TableNode $table)
+    {
+        $data = $this->extractFirstHorizontalRowFromProperties($table);
+        $expectedEditableContact = $this->mapToEditableContact($contactId, $data);
+        /** @var EditableContact $editableContact */
+        $editableContact = $this->getQueryBus()->handle(new GetContactForEditing($contactId));
+        PHPUnit_Framework_Assert::assertEquals($expectedEditableContact, $editableContact);
```

Comment:
I'm not sure we should mix Behat and PHPUnit

Reply:
getRowsHash() is for vertical table; As I continue to develop Behat tests I agree it is a better approach to use vertical tables (even sometimes horizontal one looks nicer).  Thank you for the notice.

Reply:
Thank you jolelievre for

> This seems really overkill, doesn't a simple cast work? (bool) $contact->customer_service The only use case that the class can handle is if the string is equal to 'false' But I don't think the legacy class ever return this and expect it to be considered false
> As for the behat values then we should simply use 0

The simple cast works. And for Behat tests I have found nice Ferment function in PrimiteUtils

```
    public static function castStringBooleanIntoBoolean($element)
    {
        if ($element === 'false') {
            return false;
        }

        return boolval($element);
    }

The only case it could be needed when the 'false' comes from the Request to Controller (for example from the checkbox). And if Symfony forms cover this case then it is really not needed.
```

Reply:
@jolelievre we are only using PHPUnit assertion functions when they are useful. So far in our behat tests we have been using homemade assertion logic and throwing exceptions when the  result was bad ; PHPUnit provides some nice assertions that can save us from writing a lot of boilerplate code.

I've seen it multiple times 😊for projects using behat: using PHPUnit not as a unit test engine but as library full of useful assertions.